crates/bevy_transform/src/components/children.rs

```diff
@@ -26,15 +26,9 @@ impl Children {
 }
 
 impl Deref for Children {
-    type Target = SmallVec<[Entity; 8]>;
+    type Target = [Entity];
 
     fn deref(&self) -> &Self::Target {
-        &self.0
-    }
-}
-
-impl DerefMut for Children {
```

Comment:
we need some way to re-order children. order already matters for things like UI, and once we have an editor it will matter for "scene organization".

I added this to enable those scenarios:

```rs
children.swap(a_index, b_index);
```